### ISSUE TITLE
ImagePaint unexcpectedly resizes image half automatically when used as output.

### ISSUE THREAD

**Issue by TkskKurumi** [gradio-app/gradio#3138]
### Describe the bug

Hi experts,

When I use an Image component as image uploading and sync (tested both with Image.upload or Button.submit) its image to an ImagePaint component, the image in the paint component iterates with image is resized half in each iteration, the .edit event is called without any user operation. Sorry I'm not good at expressing the problem but if you reproduce with the code below I think it'll be clear.

Thanks.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python3
import gradio as gr
def debug(img):
    print(img.shape) # found that Image is resized halfsize many times until finaly tiny blur image
    return None
with gr.Blocks() as demo:
    
        
        im_orig = gr.Image(source="upload")
        im_paint = gr.ImagePaint()
    
        im_orig.upload(lambda x: x, inputs=im_orig, outputs=im_paint) # or use a button and with btn.submit

        im_paint.edit(debug, inputs=im_paint, outputs=[])
        
demo.launch(debug=True)
```

### Screenshot

![image](https://user-images.githubusercontent.com/29892616/217129809-e550b628-26d2-4453-8b5e-ea10ba824880.png)

### Logs

```shell
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
(494, 494, 3)
(201, 201, 3)
(64, 64, 3)
(20, 20, 3)
(5, 5, 3)
```

### System Info

```shell
Gradio: 3.17.1
Python: Python 3.10.8 | packaged by conda-forge | (main, Nov  4 2022, 13:42:51) [MSC v.1916 64 bit (AMD64)] on win32
Browser: Chrome 109.0.5414.120
```

### Severity

annoying

**Comment by freddyaboulton** [comment]
Yea I can reproduce. The whole app slows down and becomes unusable regardless of whether the `edit` event is attached to `im_paint`

**Comment by abidlabs** [comment]
I was hoping this was fixed as part of PR #3277 but I just tested and this is still an issue (just fyi @pngwn)
